src/Lucene.Net/Index/DocumentsWriter.cs

```diff
@@ -1269,13 +1269,15 @@ internal bool ApplyDeletes(SegmentInfos infos, IState state)
 				
 				int docStart = 0;
 				bool any = false;
+                writer.poolReaders = true;
```

Comment:
Can you please tell more about that change? I just have little understanding based on https://issues.hibernatingrhinos.com/issue/RDBS-124#focus=Comments-67-55931.0-0

I see that this field is also used in `IndexWriter.MergeMiddle()`. This means that we're affecting the merge behavior as well.  Are we okay with that?